README.md

```diff
@@ -96,61 +93,67 @@ our [official documentation](https://docs.conjur.org/Latest/en/Content/Developer
+ssl_verification_mode = SslVerificationMode.TRUST_STORE
+```
 
-client = Client(conjurrc_data=ConjurrcData(...),
-                ssl_verification_mode=SslVerificationMode.TRUST_STORE,
-                credentials_provider=FileCredentialsProvider(),
-                debug=False)
+#### Define ConjurrcData
 
-print("Setting variable...")
-client.set('conjur/my/variable', 'new value')
+ConjurrcData is a data class containing all the non-credentials connection details.
 
-print("Fetching variable...")
-new_value = client.get('conjur/my/variable')
+`connection_info = ConjurrcData(conjur_url=conjur_url,account=account,cert_file = None)`
 
-print("Variable value is:", new_value.decode('utf-8'))
-```
+* conjur_url - url of conjur server
+* account - the account which we want to connect to
+* cert_file - a path to conjur rootCA file. we need it if we initialize the client in `SslVerificationMode.SELF_SIGN`
+  or `SslVerificationMode.CA_BUNDLE` mode
 
-#### Defining the Conjur server endpoint
+#### Storing credentials
 
-A configuration file called `.conjurrc` is used to hold details required to communicate to the
-Conjur server. You can provide these details needed to open a connection to the Conjur endpoint in
-this file instead of passing them in (`url`, `account`, and `ca_bundle`)  
-during initialization of the Client.
+The client uses credentials provider called `CredentialStores` which inherit from `CredentialsStoreInterface`. This
+approach allow to keep the credentials in a safe location and provide it to the client on demand.
 
-The `.conjurrc` file should be saved to your home directory and should contain `conjur_url`,
-`conjur_account`, and`cert_file`.
+We provide the user with `CredentialStoreFactory` which create such Credential stores.
 
+By default, the `CredentialStoreFactory` will favor saving credentials (login ID and password) to the system's
+credential store. If the detected credential store is not one we support or is not accessible, the credentials will be
+written to a configuration file, `.netrc`, in plaintext.
+
+Example of usage:
+##### In case of first connection to conjur
 ```
-# .conjurrc
----
-cert_file: /Users/someuser/conjur-server.pem
-conjur_account: someaccount
-conjur_url: https://conjur-server
-```
+credentials = CredentialsData(login=username, password=api_key, machine=conjur_url)
 
-#### Storing credentials
+credentials_provider = CredentialStoreFactory.create_credential_store()
+
+credentials_provider.save(credentials)
 
-When using the SDK, you can keep credentials in the system's native credential store instead of
-passing them in during initialization of the Client. By default, the Client will favor saving
-credentials (login ID and password) to the system's credential store. If the detected credential
-store is not one we support or is not accessible, the credentials will be written to a configuration
-file, `.netrc`, in plaintext.
+del credentials
+```
+Note: The password should be in the form of the api_key.
```

Comment:
what does this mean?

Reply:
Do you mean " The password is the api_key, for example 12345jhgdkfhgkh5345h4ifyaife4p9hwg?

Reply:
yep

Reply:
So then 
```suggestion
Note: The password is the api_key, for example ....(@mbenita-Cyberark  give a dummy example of an api key)
```